src/tasks/laika-sync.ts

```diff
@@ -11,27 +11,57 @@ import { endpointUrls } from "../config";
  * @param {string} contract - The name of the contract you want to sync.
  * @param {string} contractAddress - The Address of that specific contract.
  */
-const laikaSync = async (hre: HardhatRuntimeEnvironment, contract: string, contractAddress: string) => {
-  const { abi } = await hre.artifacts.readArtifact(contract);
-  console.log(`Syncing the ABI of ${contract} contract...`);
-
-  const { default: fetch } = await import("node-fetch");
-  const response = await fetch(
-    `${endpointUrls.services}/abi-storages`,
-    {
-      method: "POST",
-      body: JSON.stringify({ abi, contractAddress }),
-      headers: { "Content-Type": "application/json" },
-    }
-  );
+const laikaSync = async (
+  hre: HardhatRuntimeEnvironment,
+  contract: string,
+  contractAddress: string
+) => {
+  if (!contract) {
+    throw new NomicLabsHardhatPluginError(
+      "hardhat-laika",
+      "Some parameters are missing"
+    );
+  }
+
+  const contractNames = contract.split(",");
+  const contractAddresses = contractAddress?.split(",");
+
+  const endpoints = [];
 
-  const publicUrl = await response.text();
-  const endpoint = `${endpointUrls.interface}/evm/collections/import/${
-    publicUrl.split("/")[4].split(".")[0]
-  }`;
+  for (let i = 0; i < contractNames.length; i++) {
+    try {
+      const { abi } = await hre.artifacts.readArtifact(contractNames[i]);
+      console.log(`Syncing the ABI of ${contractNames[i]} contract...`);
+
+      const { default: fetch } = await import("node-fetch");
+      const response = await fetch(`${endpointUrls.services}/abi-storages`, {
+        method: "POST",
+        body: JSON.stringify({
+          abi,
+          contractAddress: contractAddresses?.[i],
+        }),
+        headers: { "Content-Type": "application/json" },
+      });
+      const publicUrl = await response.text();
+
+      endpoints.push(publicUrl.split("/")[4].split(".")[0]);
+    } catch {
+      console.log(`Fail to sync the ABI of ${contractNames[i]} contract`);
```

Comment:
this should throw out `HardhatPluginError`

ref: https://hardhat.org/hardhat-runner/docs/advanced/building-plugins#throwing-errors-from-your-plugins

Reply:
I think if it successfully uploads only one contract, it should continue working instead of throwing an error.

Reply:
understandable, but i think it will be better if users see reasons why it's not working na? `console.error` is fine i guess